openassessment/xblock/static/js/src/studio/oa_edit_settings.js

```diff
@@ -27,11 +28,38 @@ OpenAssessment.EditSettingsView = function(element, assessmentViews) {
         "#openassessment_submission_due_time"
     ).install();
 
+    new OpenAssessment.SelectControl(
+        $("#openassessment_submission_upload_selector", this.element),
+        {'custom': $("#openassessment_submission_white_listed_file_types_wrapper", this.element)},
+        new OpenAssessment.Notifier([
+            new OpenAssessment.AssessmentToggleListener()
+        ])
+    ).install();
+
     this.leaderboardIntField = new OpenAssessment.IntField(
         $("#openassessment_leaderboard_editor", this.element),
         { min: 0, max: 100 }
     );
 
+    this.fileTypeWhiteListInputField = new OpenAssessment.InputControl(
+        $("#openassessment_submission_white_listed_file_types", this.element),
+        function(value) {
+            var badExts = [];
+            var errors = [];
+            var whiteList = $.map(value.split(','), $.trim);
+            $.each(whiteList, function(index, ext) {
+                if (data.FILE_EXT_BLACK_LIST.indexOf(ext) !== -1) {
+                    badExts.push(ext);
+                }
+            });
+            if (badExts.length) {
+                errors.push(gettext('The following file types are not allowed: ') + badExts.join(','));
+            }
```

Comment:
I think that stripping periods if present as well as changing the input to lower case will make for a better user experience. Thanks for making this change, @xcompass .